features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Configuration/components/StoreVisibility.jsx

```diff
@@ -123,17 +164,49 @@ export default function StoreVisibility(props) {
             </Grid>
             {!isPublic && (
                 <Grid item>
-                    <TextField
-                        label='Role(s)'
-                        margin='dense'
-                        variant='outlined'
-                        value={api.visibleRoles.join(',')}
-                        onChange={({ target: { value } }) => configDispatcher({ action: 'visibleRoles', value })}
+                    <ChipInput
+                        value={api.visibleRoles.length === 0 ? invalidRoles : api.visibleRoles.concat(invalidRoles)}
+                        alwaysShowPlaceholder={false}
+                        placeholder='Enter roles and press Enter'
+                        blurBehavior='clear'
+                        InputProps={{
+                            endAdornment: !roleValidity && (
+                                <InputAdornment position='end'>
+                                    <Error color='error' />
+                                </InputAdornment>
+                            ),
+                        }}
+                        onAdd={(role) => {
+                            handleRoleAddition(APIValidation.role.validate(base64url.encode(role)), role);
```

Comment:
Shall we move this `APIValidation.role.validate(base64url.encode(role))` call inside the `handleRoleAddition ` function? So it will look more clearner.

Reply:
Fixed in f20d89587c00c4aadaf582e15f4f469743872571